pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -90,16 +90,23 @@ func (r Requirements) Add(requirements ...v1.NodeSelectorRequirement) Requiremen
 			continue
 		}
 		r.Requirements = append(r.Requirements, requirement)
+		var values sets.Set
 		switch requirement.Operator {
 		case v1.NodeSelectorOpIn:
-			r.requirements[requirement.Key] = r.Get(requirement.Key).Intersection(sets.NewSet(requirement.Values...))
+			values = sets.NewSet(requirement.Values...)
 		case v1.NodeSelectorOpNotIn:
-			r.requirements[requirement.Key] = r.Get(requirement.Key).Intersection(sets.NewComplementSet(requirement.Values...))
+			values = sets.NewComplementSet(requirement.Values...)
 		case v1.NodeSelectorOpExists:
-			r.requirements[requirement.Key] = r.Get(requirement.Key).Intersection(sets.NewComplementSet())
+			values = sets.NewComplementSet()
 		case v1.NodeSelectorOpDoesNotExist:
-			r.requirements[requirement.Key] = sets.NewSet()
+			values = sets.NewSet()
+		default:
+			values = sets.NewSet()
```

Comment:
This default case should never happen given what we currently allow, right?

Reply:
Correct. I'll rip this out. I only needed it when this was a helper function.